app/Helpers/database/achievement.php

```diff
@@ -27,55 +27,77 @@ function getAchievementsList(
     $bindings = [
         'offset' => $offset,
         'limit' => $limit,
+        'achievementFlag' => $achievementFlag,
     ];
 
-    $innerJoinPlayerAchievements = "";
-    $withAwardedDate = "";
-    if ($params > 0 && $user) {
-        $innerJoinPlayerAchievements = "LEFT JOIN player_achievements AS pa ON pa.achievement_id = ach.ID AND pa.user_id = " . $user->id;
-        $withAwardedDate = ", COALESCE(pa.unlocked_hardcore_at, pa.unlocked_at) AS AwardedDate";
-    }
+    $selectAwardedDate = ", NULL AS AwardedDate";
+    $joinPlayerAchievements = "";
+    $additionalWhereClauses = "";
+    $notExistsSubquery = "";
 
     // We can't run a sort on a user's achievements AwardedDate
     // if we don't have a user. Bail from the sort.
     if (($sortBy == 9 || $sortBy == 19) && !$user) {
         $sortBy = 0;
     }
 
+    if ($params === 1) {
+        // Achievements the user has unlocked.
+        $bindings['userId'] = $user->id;
```

Comment:
Does this need a null check? `$user` is nullable, but this function only appears to be called from `achievementList.blade.php`, which forces `$params` to 3 if `$user` is null.

Reply:
Doesn't hurt to add. The function probably shouldn't be at the mercy of its caller to pass args correctly. Added an `isset()` to the conditional in latest.